### PR TITLE
Adding newer 6th order mass-lumped triangle

### PR DESCRIPTION
**Pull request by Olender**
This PR introduces a new 6th-order mass-lumped triangular element (KMV). This element improves memory storage for some Full Waveform Inversion (FWI) cases.

### CODE REVIEW

File: FIAT/quadrature_schemes.py [hunk 1]
```diff
@@ -329,6 +329,133 @@ def _kmv_lump_scheme(ref_el, degree):
...
+            alpha = alphas[1]
+            x.extend(
+                [
+                    (1 - alpha, alpha),
+                    (alpha, 1 - alpha),
+                    (0.0, 1 - alpha),
+                    (0.0, alpha),
+                    (alpha, 0.0),
+                    (1 - alpha, 0.0),
+                ]  # edge points (1st set of class 3 points)
```
Comment (pbrubeck) [inline FIAT/quadrature_schemes.py:382]: We should consider simplifying/generalizing this code. For instance I see that this code generates points on each individual edge for each degree. We could generate the points on the interval and then transform them with something like, 

```
pts = []
edge_points = generate_edge_kmv_points(degree)
top = ref_el.get_topololgy()
for edge_id in sorted(top[1])
   transform = ref_el.get_entity_transform(1, edge_id)
   pts.extend(transform(edge_points))
```
Also, do we know if the edge points coincide with any well known family of points, say GLL?

File: FIAT/quadrature_schemes.py [hunk 2]
```diff
@@ -329,6 +329,133 @@ def _kmv_lump_scheme(ref_el, degree):
...
+                ]  # interior points on bisector (2nd set of class 5 points)
+            )
+            beta = betas[2]
+            x.extend(
+                [
+                    (beta, beta),
+                    (1 - 2 * beta, beta),
+                    (beta, 1 - 2 * beta)
+                ]  # interior points on bisector (3rd set of class 5 points)
+            )
```
Comment (pbrubeck) [inline FIAT/quadrature_schemes.py:419]: This block is could use a for loop, and maybe this also true for the other degrees. It'd be nice if we can write all of this in an unified way for every degree.

File: FIAT/quadrature_schemes.py [hunk 3]
```diff
@@ -329,6 +329,133 @@ def _kmv_lump_scheme(ref_el, degree):
...
+            x.extend(
+                [
+                    (gamma, delta),
+                    (1 - gamma - delta, delta),
+                    (gamma, 1 - gamma - delta),
+                    (delta, gamma),
+                    (1 - gamma - delta, gamma),
+                    (delta, 1 - gamma - delta),
+                ]  # interior points (2st set of class 6 points)
+            )
```
Comment (pbrubeck) [inline FIAT/quadrature_schemes.py:444]: same here

File: FIAT/kong_mulder_veldhuizen.py [hunk 4]
```diff
@@ -77,6 +78,14 @@ def _get_entity_ids(ref_el, degree):
...
                 2: {0: [i for i in range(15, 30)]},
             }
+    elif degree == 6:
+        if sd == 2:
+            etop = [[6, 12, 3, 13, 7], [9, 15, 4, 14, 8], [10, 16, 5, 17, 11]]
+            entity_ids = {
+                0: dict((i, [i]) for i in range(3)),
+                1: dict((i, etop[i]) for i in range(3)),
+                2: {0: [i for i in range(18, 39)]},
+            }
```
Comment (pbrubeck) [inline FIAT/kong_mulder_veldhuizen.py:88]: Can we improve the ordering here? Say we always want to order vertex dofs first, followed by all dofs of edge 0, edge1, edge2, and finally the interior dofs.

File: FIAT/quadrature_schemes.py (hunk 1 above)
Comment (Olender) [reply FIAT/quadrature_schemes.py:382]: Unfortunately, for degrees 3 and up, the edge points do not coincide with any well-known family of points, such as GLL, GRL, etc.. These elements are constructed by searching for configurations that satisfy specific geometric constraints, accuracy conditions, and positive weights, but no general rule is established yet. That’s why I’m introducing this new 6th-order element. It's entirely possible that, in the future, better elements for various orders could be discovered by refining or altering these search methods.

File: FIAT/quadrature_schemes.py (hunk 3 above)
Comment (Olender) [reply FIAT/quadrature_schemes.py]: Thank you for the suggestion! I’ve modified degrees 4, 5, and 6 to use for loops for the repeated point sets to improve readability and reduce redundancy. I considered further unifying the structure for all degrees, but due to slight structural differences, a fully unified approach might make the code more complex.

For example, degrees 3, 4, 5, and 6 share the edge points symmetrically using the alpha variable and have interior points along the bisector using the beta variable. However, only degrees 5 and 6 include additional interior points determined by the gamma and delta variables

Comment (pbrubeck) [reply FIAT/quadrature_schemes.py]: Thanks

**Review by pbrubeck — APPROVED** [review]
Looks good

**Pull request merged**